### ISSUE TITLE
assertion failed on a simple circuit

### ISSUE THREAD

**Issue by themighty1** [zcash/halo2#651]
Hi, I'm stuck on having a simple circuit: a public input == 2 and a private input == 2 and a simple gate constraining with `vec![sel * (private - public)]`
I keep getting an error `thread 'main' panicked at 'assertion failed: `(left == right)``

I'd be grateful if you could unblock me.

Below is a minimal repro:

```
use halo2_proofs::{
    circuit::{Layouter, SimpleFloorPlanner, Value},
    dev::MockProver,
    plonk::{Advice, Circuit, Column, ConstraintSystem, Error, Instance, Selector},
    poly::Rotation,
};
use pasta_curves::arithmetic::FieldExt;
use pasta_curves::pallas;

type F = pallas::Base;

#[derive(Clone, Debug)]
struct TopLevelConfig {
    advice: Column<Advice>,
    instance: Column<Instance>,
    selector: Selector,
}

struct MyCircuit {
    private: Value<F>,
}

impl Circuit<pallas::Base> for MyCircuit {
    type Config = TopLevelConfig;
    type FloorPlanner = SimpleFloorPlanner;

    fn without_witnesses(&self) -> Self {
        Self {
            private: Value::unknown(),
        }
    }

    fn configure(meta: &mut ConstraintSystem<pallas::Base>) -> Self::Config {
        let advice = meta.advice_column();
        meta.enable_equality(advice);

        let selector = meta.selector();

        let instance = meta.instance_column();
        meta.enable_equality(instance);

        meta.create_gate("gate", |meta| {
            let private = meta.query_advice(advice, Rotation::cur());
            let public = meta.query_instance(instance, Rotation::cur());
            let sel = meta.query_selector(selector);

            vec![sel * (private - public)]
        });

        TopLevelConfig {
            advice,
            instance,
            selector,
        }
    }

    fn synthesize(
        &self,
        config: Self::Config,
        mut layouter: impl Layouter<F>,
    ) -> Result<(), Error> {
        layouter
            .assign_region(
                || "main",
                |mut region| {
                    region.assign_advice(|| "", config.advice, 0, || self.private)?;
                    config.selector.enable(&mut region, 0)?;
                    Ok(())
                },
            )
            .unwrap();
        Ok(())
    }
}

pub fn main() {
    let k = 7;

    let public = vec![vec![pallas::Base::from_u128(2u128)]];

    let circuit: MyCircuit = MyCircuit {
        private: Value::known(pallas::Base::from_u128(2u128)),
    };

    let prover = MockProver::<pallas::Base>::run(k, &circuit, public).unwrap();

    assert_eq!(prover.verify(), Ok(()));
}

```

**Comment by therealyingtong** [comment]
I think this is https://github.com/zcash/halo2/issues/621:

```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `Err([CellNotAssigned { gate: Gate { index: 0, name: "gate" }, region: Region { index: 0, name: "main" }, gate_offset: 0, column: Column { index: 0, column_type: Instance }, offset: 0 }])`,
 right: `Ok(())`'
```

**Comment by therealyingtong** [comment]
Closing as a duplicate of #621.
